File: dotNET/PFProSDK/src/PayPal/Payments/Transactions/BuyerAuthTransaction.cs
#region "Imports"

using System;
using PayPal.Payments.DataObjects;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.Transactions
{
	/// <summary>
	/// This abstract class serves as base class for 
	/// Buyer auth transactions.
	/// </summary>
	
	public class BuyerAuthTransaction : BaseTransaction
	{
		#region "Constructors"

		/// <summary>
		/// protected Constructor. This prevents
		/// creation of an empty Transaction object. 
		/// </summary>
		protected BuyerAuthTransaction()
		{
		}

		/// <summary>
		/// Constructor.
		/// </summary>
		/// <param name="TrxType">Transaction type</param>
		/// <param name="UserInfo">User Info object populated with user credentials.</param>
		/// <param name="PayflowConnectionData">Connection credentials object.</param>
		/// <param name="RequestId">Request Id</param>
		protected BuyerAuthTransaction(String TrxType, UserInfo UserInfo, PayflowConnectionData PayflowConnectionData, String RequestId)
			: base(TrxType, UserInfo, PayflowConnectionData, RequestId)
		{
		}
		/// <summary>
		/// Constructor.
		/// </summary>
		/// <param name="TrxType">Transaction type</param>
		/// <param name="UserInfo">User Info object populated with user credentials.</param>
		/// <param name="RequestId">Request Id</param>
		protected BuyerAuthTransaction(String TrxType, UserInfo UserInfo, String RequestId)
			: this(TrxType, UserInfo, null, RequestId)
		{
		}

		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			try
			{
				base.GenerateRequest();
			}
			catch (BaseException)
			{
				throw;
			}
			catch (Exception Ex)
			{
				TransactionException TE = new TransactionException(Ex);
				throw TE;
			}
            //catch
            //{
            //    throw new Exception();
            //}
		}

		#endregion
	}

}

File: java/src/sdk/transactions/paypal/payflow/BuyerAuthTransaction.java
/*
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package paypal.payflow;

/**
 * This abstract class serves as base class for
 * Buyer auth transactions.
 */
public class BuyerAuthTransaction extends BaseTransaction {

    /**
     * @param trxType               Transaction type
     * @param userInfo              User Info object populated with user credentials.
     * @param payflowConnectionData Connection credentials object.
     * @param requestId             String
     */
    protected BuyerAuthTransaction(String trxType, UserInfo userInfo, PayflowConnectionData payflowConnectionData, String requestId) {
        super(trxType, userInfo, payflowConnectionData, requestId);
    }

    /**
     * @param trxType   Transaction type
     * @param userInfo  User Info object populated with user credentials.
     * @param requestId String
     */
    protected BuyerAuthTransaction(String trxType, UserInfo userInfo, String requestId) {
        this(trxType, userInfo, null, requestId);
    }
}

